File: day3/part1/solution.py
import os

def solve():
    """
    Solves the Day 3, Part 1 puzzle.
    """
    input_path = os.path.join(os.path.dirname(__file__), 'input.txt')
    with open(input_path, 'r') as f:
        lines = f.readlines()

    total_joltage = 0
    for line in lines:
        line = line.strip()
        if not line:
            continue
        
        max_line_joltage = 0
        for i in range(len(line)):
            for j in range(i + 1, len(line)):
                joltage_str = line[i] + line[j]
                joltage = int(joltage_str)
                if joltage > max_line_joltage:
                    max_line_joltage = joltage
        total_joltage += max_line_joltage
        
    print(f"The total output joltage is: {total_joltage}")

if __name__ == "__main__":
    solve()

File: day3/part1/solution.rb
def solve
  input_path = File.join(File.dirname(__FILE__), 'input.txt')
  lines = File.readlines(input_path)

  total_joltage = 0
  lines.each do |line|
    line = line.strip
    next if line.empty?

    max_line_joltage = 0
    (0...line.length).each do |i|
      (i + 1...line.length).each do |j|
        joltage_str = line[i] + line[j]
        joltage = joltage_str.to_i
        if joltage > max_line_joltage
          max_line_joltage = joltage
        end
      end
    end
    total_joltage += max_line_joltage
  end

  puts "The total output joltage is: #{total_joltage}"
end

solve
